prov/rxm/src/rxm.h

```diff
@@ -474,6 +473,16 @@ struct rxm_tx_rndv_buf {
 	struct rxm_pkt pkt;
 };
 
+union rxm_tx_buf {
+	struct {
+		struct rxm_buf hdr;
+		enum rxm_buf_pool_type type;
```

Comment:
Can this be moved inside rxm_buf?

Reply:
I think we could remove this field at all, because we can identify the buffer pool from which this TX buffer was allocated.